### ISSUE TITLE
debugger crash

### ISSUE THREAD

**Issue by aler** [JakeBecker/elixir-ls#124]
### Environment

* Erlang/OTP 21 [erts-10.0.8] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:1] [hipe] [dtrace] Elixir 1.7.3 (compiled with Erlang/OTP 21) (installed using brew install elixir)
* MacOs 10.14
* VsCode 1.28

### Crash report template

```
Started ElixirLS debugger
(Debugger) Terminating because an exception was raised:
    ** (FunctionClauseError) no function clause matching in ElixirLS.Debugger.Server.handle_request/2
        (debugger) lib/debugger/server.ex:123: ElixirLS.Debugger.Server.handle_request(%{"arguments" => %{"filters" => []}, "command" => "setExceptionBreakpoints", "seq" => 3, "type" => "request"}, %ElixirLS.Debugger.Server{breakpoints: %{}, client_info: %{"adapterID" => "mix_task", "clientID" => "vscode", "clientName" => "Visual Studio Code", "columnsStartAt1" => true, "linesStartAt1" => true, "locale" => "en-us", "pathFormat" => "path", "supportsRunInTerminalRequest" => true, "supportsVariablePaging" => true, "supportsVariableType" => true}, config: %{"__sessionId" => "cc4afc75-70a2-499c-883f-0fdbd411f5ab", "name" => "mix test", "projectDir" => "/Users/alex/Repos/alr/counter", "request" => "launch", "requireFiles" => ["test/**/test_helper.exs", "test/**/*_test.exs"], "startApps" => true, "task" => "test", "taskArgs" => ["--trace"], "type" => "mix_task"}, next_id: 1, output: ElixirLS.Debugger.Output, paused_processes: %{}, task_ref: nil, threads: %{}, threads_inverse: %{}})
        (debugger) lib/debugger/server.ex:62: ElixirLS.Debugger.Server.handle_cast/2
        (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
        (stdlib) gen_server.erl:711: :gen_server.handle_msg/6
        (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
19:14:27.220 [error] GenServer ElixirLS.Debugger.Server terminating
** (FunctionClauseError) no function clause matching in ElixirLS.Debugger.Server.handle_request/2
    (debugger) lib/debugger/server.ex:123: ElixirLS.Debugger.Server.handle_request(%{"arguments" => %{"filters" => []}, "command" => "setExceptionBreakpoints", "seq" => 3, "type" => "request"}, %ElixirLS.Debugger.Server{breakpoints: %{}, client_info: %{"adapterID" => "mix_task", "clientID" => "vscode", "clientName" => "Visual Studio Code", "columnsStartAt1" => true, "linesStartAt1" => true, "locale" => "en-us", "pathFormat" => "path", "supportsRunInTerminalRequest" => true, "supportsVariablePaging" => true, "supportsVariableType" => true}, config: %{"__sessionId" => "cc4afc75-70a2-499c-883f-0fdbd411f5ab", "name" => "mix test", "projectDir" => "/Users/alex/Repos/alr/counter", "request" => "launch", "requireFiles" => ["test/**/test_helper.exs", "test/**/*_test.exs"], "startApps" => true, "task" => "test", "taskArgs" => ["--trace"], "type" => "mix_task"}, next_id: 1, output: ElixirLS.Debugger.Output, paused_processes: %{}, task_ref: nil, threads: %{}, threads_inverse: %{}})
    (debugger) lib/debugger/server.ex:62: ElixirLS.Debugger.Server.handle_cast/2
    (stdlib) gen_server.erl:637: :gen_server.try_dispatch/4
    (stdlib) gen_server.erl:711: :gen_server.handle_msg/6
    (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: {:"$gen_cast", {:receive_packet, %{"arguments" => %{"filters" => []}, "command" => "setExceptionBreakpoints", "seq" => 3, "type" => "request"}}}
ElixirLS debugger has crashed
```

### dev tools console log

```
  ERR Cannot read property 'threads' of undefined: TypeError: Cannot read property 'threads' of undefined
    at e.fetchThreads (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2969:99)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2969:716
    at Object.g [as _notify] (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:160:729)
    at Object.enter (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:164:568)
    at n.Class.derive._creator._run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:902)
    at n.Class.derive._creator._error (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:419)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:163:75)
    at n.Class.define.cancel.then (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:166:736)
    at Object.enter (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:163:109)
    at n.Class.derive._creator._run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:902)
    at n.Class.derive._creator._setCompleteValue (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:596)
    at Object.m [as _notify] (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:161:284)
    at Object.enter (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:164:568)
    at n.Class.derive._oncancel._run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:902)
    at n.Class.derive._oncancel._error (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:419)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2605:848
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2609:288
    at Map.forEach (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2609:266
errors.ts:48 WARNING: Promise with no error callback:289
errors.ts:49 {exception: TypeError: Cannot read property 'threads' of undefined
    at e.fetchThreads (file:///Applications/V…, error: null, promise: n.C…s.d…e._creator, handler: undefined, id: 289, …}error: nullexception: TypeError: Cannot read property 'threads' of undefined
    at e.fetchThreads (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2969:99)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2969:716
    at Object.g [as _notify] (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:160:729)
    at Object.enter (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:164:568)
    at n.Class.derive._creator._run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:902)
    at n.Class.derive._creator._error (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:419)
    at n (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:163:75)
    at n.Class.define.cancel.then (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:166:736)
    at Object.enter (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:163:109)
    at n.Class.derive._creator._run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:902)
    at n.Class.derive._creator._setCompleteValue (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:596)
    at Object.m [as _notify] (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:161:284)
    at Object.enter (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:164:568)
    at n.Class.derive._oncancel._run (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:902)
    at n.Class.derive._oncancel._error (file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:165:419)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2605:848
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2609:288
    at Map.forEach (<anonymous>)
    at file:///Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/workbench.main.js:2609:266handler: undefinedid: 289parent: undefinedpromise: n.Class.derive._creator {_creator: null, _nextState: null, _state: {…}, _value: TypeError: Cannot read property 'threads' of undefined
    at e.fetchThreads (file:///Applications/V…, _isException: true, …}__proto__: Object
errors.ts:51 TypeError: Cannot read property 'threads' of undefined
    at e.fetchThreads (debugSession.ts:546)
    at debugSession.ts:582
    at Object.g [as _notify] (winjs.base.js:1191)
    at Object.enter (winjs.base.js:916)
    at n.Class.derive._creator._run (winjs.base.js:1089)
    at n.Class.derive._creator._error (winjs.base.js:1062)
    at n (winjs.base.js:758)
    at n.Class.define.cancel.then (winjs.base.js:1457)
    at Object.enter (winjs.base.js:762)
    at n.Class.derive._creator._run (winjs.base.js:1089)
    at n.Class.derive._creator._setCompleteValue (winjs.base.js:1073)
    at Object.m [as _notify] (winjs.base.js:1230)
    at Object.enter (winjs.base.js:916)
    at n.Class.derive._oncancel._run (winjs.base.js:1089)
    at n.Class.derive._oncancel._error (winjs.base.js:1062)
    at rawDebugSession.ts:516
    at debugAdapter.ts:174
    at Map.forEach (<anonymous>)
    at debugAdapter.ts:165
```

### launch.json

```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "mix_task",
            "name": "mix (Default task)",
            "request": "launch",
            "projectDir": "${workspaceRoot}"
        },
        {
            "type": "mix_task",
            "name": "mix test",
            "request": "launch",
            "task": "test",
            "taskArgs": [
                "--trace"
            ],
            "startApps": true,
            "projectDir": "${workspaceRoot}",
            "requireFiles": [
                "test/**/test_helper.exs",
                "test/**/*_test.exs"
            ]
        }
    ]
}
```

Debugger crash every time I hit debug button. If you need more info I could provide elixir project source code.

**Comment by martin-watts** [comment]
I've opened #127, which fixes the problem for me, but might not be the best fix.

**Comment by JakeBecker** [comment]
I just published v0.2.24 with @martin-watts 's fix.
